### ISSUE TITLE
Bulk deletion is only partially effective

### ISSUE THREAD

**Issue by choucmei** [ClickHouse/ClickHouse#7124]
when i executed this sql.In theory, there isn't same ids in  table 'test_01' and 'test_01_id'
`alter table test_01 delete  where id in (select id from test_01_id) t`
but has many same id in table 'test_01' and 'test_01_id'
`select * from test_01 where id in (select id from test_01_id) t`
![图片](https://user-images.githubusercontent.com/32946731/65739171-f526c680-e116-11e9-9c0a-2017af9f6525.png)
and when i executed sql  this,and all data has been deleted
`alter table test_01 delete where id in (select id from test_01)`

**Comment by filimonov** [comment]
Hard to follow. 

> and when i executed sql this,and all data has been deleted
> `alter table test_01 delete where id in (select id from test_01)`

I would also say it should delete everything :)

Can you please provide some simple/synthetic example? We need simple table(s) definition, some data and queries which work not as expected.

**Comment by choucmei** [comment]
> Hard to follow.
> 
> > and when i executed sql this,and all data has been deleted
> > `alter table test_01 delete where id in (select id from test_01)`
> 
> I would also say it should delete everything :)
> 
> Can you please provide some simple/synthetic example? We need simple table(s) definition, some data and queries which work not as expected.



> Hard to follow.
> 
> > and when i executed sql this,and all data has been deleted
> > `alter table test_01 delete where id in (select id from test_01)`
> 
> I would also say it should delete everything :)
> 
> Can you please provide some simple/synthetic example? We need simple table(s) definition, some data and queries which work not as expected.
May be this error
 [ 12 ] {} <Error> : Code: 40, e.displayText() = DB::Exception: Checksums of parts don't match: hash of uncompressed files doesn't match, uncompressed hash of compressed files doesn't match (version 19.13.3.26). Data after mutation is not byte-identical to data on another replicas. We will download merged part from replica to force byte-identical result.

I got this info from /log/clickhouse-server.log

**Comment by filimonov** [comment]
Can't reproduce:
```
create table test_01 engine=MergeTree ORDER BY number AS SELECT * FROM numbers(1000);

create table test_01_d engine=MergeTree ORDER BY number AS SELECT * FROM numbers(900);

alter table test_01 delete where number in (select number from test_01_d);

SELECT count()
FROM test_01

┌─count()─┐
│     100 │
└─────────┘
```

Please check your query it contain the silly error, which makes the clickhouse behavior correct. 

```diff
-alter table test_01 delete where id in (select id from test_01)
+alter table test_01 delete where id in (select id from test_01_d)
```
